config-model-api/src/main/java/com/yahoo/config/application/api/Bcp.java

```diff
@@ -0,0 +1,122 @@
+ * A and B will now handle 150 each, but will each aim to handle 300 each in case the other goes down.
+ *
+ * @author bratseth
+ */
+public class Bcp {
+
+    private static final Bcp empty = new Bcp(List.of());
+
+    private final List<Group> groups;
+
+    public Bcp(List<Group> groups) {
+        totalMembershipSumsToOne(groups);
+        this.groups = List.copyOf(groups);
+    }
+
+    public List<Group> groups() { return groups; }
+
+    /** Returns the set of regions declared in the groups of this. */
+    public Set<RegionName> regions() {
+        return groups.stream().flatMap(group -> group.members().stream()).map(member -> member.region()).collect(Collectors.toSet());
+    }
+
+    public boolean isEmpty() { return groups.isEmpty(); }
+
+    /** Returns this bcp spec, or if it is empty, the given bcp spec. */
+    public Bcp orElse(Bcp other) {
+        return this.isEmpty() ? other : this;
+    }
+
+    private void totalMembershipSumsToOne(List<Group> groups) {
+        Map<RegionName, Double> totalMembership = new HashMap<>();
+        for (var group : groups) {
+            for (var member : group.members())
+                totalMembership.compute(member.region(), (__, fraction) -> fraction == null ? member.fraction()
+                                                                                            : fraction + member.fraction());
+        }
+        for (var entry : totalMembership.entrySet()) {
+            if (entry.getValue() != 1.0)
+                throw new IllegalArgumentException("Illegal BCP spec: All regions must have total membership fractions summing to 1.0, but " +
+                                                   entry.getKey() + " sums to " + entry.getValue());
+        }
+    }
+
+    public static Bcp empty() { return empty; }
+
+    @Override
+    public String toString() {
+        if (isEmpty()) return "empty BCP";
+        return "BCP of " + groups;
+    }
+
+    public static class Group {
+
+        private final Duration deadline;
+        private final List<RegionMember> members;
+
+        public Group(List<RegionMember> members, Duration deadline) {
+            this.members = List.copyOf(members);
+            this.deadline = deadline;
+        }
+
+        public List<RegionMember> members() { return members; }
+
+        /**
+         * Returns the max time until the other regions must be able to handle the additional traffic
+         * when a region becomes unreachable, which by default is Duration.ZERO.
+         */
+        public Duration deadline() { return deadline; }
+
+        @Override
+        public String toString() {
+            return "BCP group of " + members;
+        }
+
+    }
+
+    public record RegionMember(RegionName region, double fraction) {
+
+        public RegionMember {
+            if (fraction < 0 || fraction > 1)
+                throw new IllegalArgumentException("Fraction must be a number between 0.0 and 1.0, but got " + fraction);
```

Comment:
Consider using `Validation.requireInRange`.

Reply:
Thanks for the tip. I'll do it in the next pr.